improved editbox sizing

KuroiLight opened the pull request:
changes how editboxes are sized, labels now on top instead of beside editbox; this should make editing longer texts easier. Text should also be aligned better.
![](http://i.imgur.com/tlYFsEF.png)

-Terrillyn

sirinsidiator commented:
The code looks good, but you need to make it opt in as it will cause unexpected changes in addons that already use the editbox.
I suggest you add a boolean "isExtraWide" to the properties.

merlight commented:
How about:
```lua
if type(editboxData.isMultiline) == "number then
  if editboxData.isMultiline <= 1 then -- new single-line style
  else -- new multi-line style, the number to set height
  end
else -- old behaviour
end

sirinsidiator commented:
I'm not really a fan of multipurpose properties that do not match their name.
For me the name isMultiline clearly indicates that it is a boolean, so it should not be used to contain a number.
Just add a height property if you need one.

KuroiLight commented:
makes use of `isExtraWide`
![](http://i.imgur.com/7czEhV3.png

sirinsidiator commented:
I just merged all 4 pull requests locally and the result looks good except for two issues:
![](http://i.imgur.com/KPl06RY.jpg)
The half width single line edit box is now on the same line as the label, which causes overlapping if the label is too long and there is also an empty space below the shown controls that was not there before.

KuroiLight commented:
The spacing issue is related to the changes in #36 , I'm pretty sure its caused by the offsets set when creating the controls, I'll take a look and commit to #36.

KuroiLight commented:
fixed typo, `editboxData.isHalfWidth` -> `control.isHalfWidth`
made warning icon show up above the topright corner of the editbox when `isExtraWide` is enabled to avoid overlapping
simplified structure of `if`s
![](http://i.imgur.com/HLEQLT4)

sirinsidiator commented:
I just pulled your latest changes and get the following error:
```
user:/AddOns/LibAddonMenu-2.0/LibAddonMenu-2.0/controls/editbox.lua:103: function expected instead of nil
stack traceback:
	user:/AddOns/LibAddonMenu-2.0/LibAddonMenu-2.0/controls/editbox.lua:103: in function 'LAMCreateControl.editbox'
	user:/AddOns/AutoInvite/ui/ai_enabled_fragment.lua:46: in function 'AutoInviteUI:CreateEnabledFragment'
	user:/AddOns/AutoInvite/AutoInviteUI.lua:34: in function 'AutoInviteUI.init'
	user:/AddOns/AutoInvite/AutoInvite.lua:171: in function '(anonymous)'
```
AFAIK only AutoInvite uses the settings controls outside of the settings menu (like described in the documentation) and passes a parent that is not a panel, but I would still like to support this case, as minor versions should not break compatibility. I did that once and got a really angry pm as a result. :P

Something like ```local MIN_WIDTH = parent.GetWidth and parent:GetWidth() / 10 or 0``` should do the trick.

Other than that it looks good now.
